packages/x-charts/src/ChartsVoronoiHandler/ChartsVoronoiHandler.tsx

```diff
@@ -70,7 +71,17 @@ function ChartsVoronoiHandler(props: ChartsVoronoiHandlerProps) {
       const getXPosition = getValueToPositionMapper(xScale);
       const getYPosition = getValueToPositionMapper(yScale);
 
-      const seriesPoints = data.flatMap(({ x, y }) => [getXPosition(x), getYPosition(y)]);
+      const seriesPoints = data.flatMap(({ x, y }) => {
+        const pointX = getXPosition(x);
+        const pointY = getYPosition(y);
+
+        if (pointX < left || pointX > left + width || pointY < top || pointY > top + height) {
+          return [0, 0];
```

Comment:
Not sure this is the best approach. I've tried:

- Filtering: Messes up index value logic
- Retuning `[NaN, NaN]`: Seem to go into an infinite loop
- Retuning `[(+-)Infinity, (+-)Infinity]`: In any combination it breaks because the [Delaunator](https://github.com/mapbox/delaunator/blob/302a0cde5d2ba8ce5e460f59166e19294de839c0/index.js#L52) logic uses infinity itself to check initial values, so whenever we zoom, we are always on the min or max of the values
- Returning `[0,0]`: Doesn't seem to affect anything. I've tried with multiple points combination

Reply:
🙃 I was thing [0,0] in dataspace, not svgspace. Feels dumb now. lol

I feel option 2 will be more reasonable, since we won't have to maintain two filters... One of the issue is that `Delauney.find()` will return an index, if we can't map it between the two of them easily, there wont be a lot of value in that.